api/src/server/auth.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{
```

Comment:
Can you just generally add some more comments to the bodies of these functions? The logic isn't 100% clear.